src/agoradatatools/etl/transform/immunohisto_transform.py

```diff
@@ -46,4 +124,6 @@ def immunohisto_transform(
         entry[extra_column_name] = group[extra_columns].to_dict("records")
```

Comment:
I think because I don't use `zip` or grouping very much in Python, I'm having a little trouble parsing what this `for` loop does. It would be really helpful if you could add a comment or two in/above the for loop describing it.